### ISSUE TITLE
Lost statement series in influxdb

### ISSUE THREAD

**Issue by mdaguete** [tricksterproxy/trickster#459]
Hi,

I,m testing v1.1.0 with influxdb 1.7.10 and grafana 6.7.4. It seems with queries with multiple statements per query only the first stament series is retained in cache after the first refresh. The problem seems to be in the sort call at the end of merge function, if I disable it then it seems to work if I don't change the query window.

This is the cache content in the first time ( all series values deleted on editing)

```
{"results":[
  {"statement_id":0,"series":[{"name":"memory/limit","columns":["time","mean"],"values":[[1591345200000,134217728],....
  {"statement_id":1,"series":[{"name":"memory/request","columns":["time","mean"],"values":[[1591345200000,134217728],....
  {"statement_id":2,"series":[{"name":"memory/usage","columns":["time","mean"],"values":[[1591345200000,12496896],...
  "step":30000000000,
  "extents":[{"start":"2020-06-05T10:20:00+02:00","end":"2020-06-05T11:18:55+02:00"}]
}
```

After refreshing this is the content:

```
{"results":[
  {"statement_id":0,"series":[{"name":"memory/limit","columns":["time","mean"],"values":[[1591345200000,134217728]../
  {"statement_id":1},
  {"statement_id":2}]
  ,"step":30000000000,
  "extents":[{"start":"2020-06-05T10:20:00+02:00","end":"2020-06-05T11:18:55+02:00"},{"start":"2020-06-05T11:19:00+02:00","end":"2020-06-05T11:25:55+02:00"}]
  }
```

My trickster config is:

```
[frontend]
  listen_port = 8086
[caches]
    [caches.default]
        cache_type = 'filesystem'
    [caches.default.filesystem]
        cache_path = '/tmp/trickster'
    [caches.default.index]
        flush_interval_secs = 1
        reap_interval_secs = 1
[origins]
    [origins.default]
        is_default= true
        cache_key_prefix = 'pref'
        origin_type = 'influxdb'
        hosts = ['localhost']
        origin_url = 'https://localhost:9086/'
        timeseries_retention_factor = 40960
        fastforward_ttl_secs = 2
        multipart_ranges_disabled = true
        backfill_tolerance_secs = 65
        fast_forward_disable=true
    [origins.default.tls]
        insecure_skip_verify = true
[logging]
    log_level = 'info'

```

How can I avoid this problem ?

Thanks.

**Comment by jranson** [comment]
@mdaguete thanks for the report and such a detailed analysis. Quick question - did you use 1.0 and are finding it is a new behavior in 1.1? Thanks!

**Comment by mdaguete** [comment]
@jranson Yes 1.0 the same behaviout. I'm testing a patch and I'll submit a pr after some test.

**Comment by jranson** [comment]
hey @mdaguete thanks again for #460 (what a great catch!), which was included in v1.1.2. Can you let us know that positively fixed up your corresponding issue here? Thanks!

**Comment by jranson** [comment]
closing due to no response over 3 months.
